docs/administering/active-directory.md

```diff
@@ -0,0 +1,304 @@
+
+![Screenshot of this step](https://alpha-evgl4wnivwih0k6mzxt3.sandstorm.io/azure-ad-config/14.png)
+
+- Open the certificate file in a text editor (such as Notepad).
+
+![Screenshot of this step](https://alpha-evgl4wnivwih0k6mzxt3.sandstorm.io/azure-ad-config/15.png)
+
+- Copy/paste the contents of the file into Sandstorm's SAML login configuration under "SAML cert for above provider:".
+
+![Screenshot of this step](https://alpha-evgl4wnivwih0k6mzxt3.sandstorm.io/azure-ad-config/13.png)
+
+![Screenshot of this step](https://alpha-evgl4wnivwih0k6mzxt3.sandstorm.io/azure-ad-config/16.png)
+
+- Back in the Azure AD configuration, copy the line titled "Single Sign-On Service URL". In the Sandstorm config, paste this under "SAML provider entry point URL:".
+
+![Screenshot of this step](https://alpha-evgl4wnivwih0k6mzxt3.sandstorm.io/azure-ad-config/17.png)
+
+- In the Azure AD config, check the box: "Confirm that you have configured single sign-on as
+  described above." Clik the "next" arrow and enter your email address for service alerts from
+  Microsoft. Click the checkbox to complete configuration.
+
+![Screenshot of this step](https://alpha-evgl4wnivwih0k6mzxt3.sandstorm.io/azure-ad-config/18.png)
+
+- Click the checkbox to complete configuration.
+
+![Screenshot of this step](https://alpha-evgl4wnivwih0k6mzxt3.sandstorm.io/azure-ad-config/19.png)
+
+- Click: "Users and Groups", then "All Users". Click the check-mark to request Sandstorm access for
+  all users. Note that Sandstorm only counts a user for [billing](for-work.md) if they actually log into
+  Sandstorm during a particular month.
+
+![Screenshot of this step](https://alpha-evgl4wnivwih0k6mzxt3.sandstorm.io/azure-ad-config/22.png)
+
+- The system will ask you if you are sure. Click "Yes" to confirm.
+
+![Screenshot of this step](https://alpha-evgl4wnivwih0k6mzxt3.sandstorm.io/azure-ad-config/23.png)
+
+**Congratulations!** You have enabled Sandstorm for all the users in your Microsoft Azure Active Directory service.
+
+## Windows Server Active Directory
+
+**Summary:** Once you have created a directory, you must use Active Directory Federation Services to
+add a new Relying Party, namely the Sandstorm web application. Each Sandstorm server publishes
+federation metadata information at a URL available within the SAML configuration page.  You must
+configure a rule that sends the user email address to Sandstorm, as well as a rule that declares
+email addresses are the persistent identifier. You must also export the custom SAML certificate to
+Sandstorm.  Once you have done this, you can use Active Directory to choose which users in your
+domain are allowed to use Sandstorm.
+
+To connect to Active Directory via SAML, you must enable Active Directory Federation Services (AD
+FS). We will assume you have already done this. We also assume that the Sandstorm server can
+reach the Active Directory server, e.g. they are on the same VPN, or firewall rules permit them
```

Comment:
Actually, Sandstorm does not need to reach the AD server. Only the client's browser needs to reach it. With SAML all communication between the SP and IdP is through the browser.

The only server<->server communication happens during setup, where AD FS fetches the XML config blob from Sandstorm. So it's actually Sandstorm that needs to be accessible from AD. (Though, this could be worked around by the admin downloading the XML from Sandstorm and then uploading it to AD separately.)